packages/wrangler/src/create-worker-preview.ts

```diff
@@ -118,11 +118,15 @@ async function createPreviewToken(
       : `/accounts/${accountId}/workers/scripts/${scriptId}/edge-preview`;
 
   const mode: CfPreviewMode = ctx.zone
-    ? { routes: ["*/*"] } // TODO: should we support routes here? how?
+    ? { routes: [...(ctx.zone.routes ?? "*/*")] }
     : { workers_dev: true };
 
   const formData = createWorkerUploadForm(worker);
-  formData.set("wrangler-session-config", JSON.stringify(mode));
+
+  const modeFile = new File([JSON.stringify(mode)], "", {
+    type: "application/json",
+  });
+  formData.append("wrangler-session-config", modeFile);
```

Comment:
Let's not make this change if there's no observable difference.

Reply:
> @JacobMGEvans like you'd mentioned, could you make the table comparing v1 and v2 with this PR and what the differences in behavior are?

I made the comparing table, it wont be helpful since there is not comparable behavior shared between the two. These changes also don't seem to work though we had them working at one point with naked domains. I will add you to the Google spreadsheet tomorrow.

Reply:
The only difference currently would be sending the route patterns with the request, also in the same way that the Rust implementation did it.

It may not fix the secrets issue it, however, brings the implementation closer to parity; seems like that can be separate from the secrets issue.